File: remove_dupe.py
# Remove immediate duplicates in a provided string

class Solution: # 'nuff said
    def solve(self, s): # declaires a function under the name of 'solve', which accepts the argument for s. The class declaired above allows the used of 'self' in order to tidy up and speed up our code
        temp = "" # creates an empty string named 'temp'
        new = [] # creates an empty array named 'new'
        string_form = '' # creates an empty string named 'string_form'
        for i in list(s): # this is a for loop, which will repeat itself over and over, but everytime it repeats, it will assign the next char in a list to whatever variable you desire, which in my case is 'i' being declaired a new char from s.
                          # 'list(s)' turns the string [s] into a list/array
            if i == temp: # if the current char equals the previous char, then:
                temp = i # the 'previous char' mentioned before is assigned the current char
            else: # 'nuff said
                temp = i # the 'previous char' mentioned before is assigned the current char
                new.append(temp) # if previous char does not equal the current char, then it's appended/added to the array called 'new'
        return ''.join(new) # once the for loop finishes, the array 'new' is turned back into the string form using 'join' and returned 


# Your code took 5 milliseconds — faster than 95.92% in Python :: https://binarysearch.com/problems/Consecutive-Duplicates

File: remove_dupe.java
// Remove immediate duplicates in a provided string.

import java.util.*; // important import of the collection framework : ect/.

class Solution { // 'nuff said
    public String solve(String s) { // sets the argument included to the variable s in the string type.
        StringBuilder result = new StringBuilder(); // Creates the array result, which will be converted into a string representation when called upon.
        char lastChar = Character.UNASSIGNED; // Character contains a single field, which is typed char. The Unassigned tag clears the single field of Character.
        for (char currentChar : s.toCharArray()) { // this is a for loop, which declairs a new variable: 'currentChar' the value of the current character selected in the char array.
                                                   // The 's.toCharArray' turns the string s, which is declaired in the public statement above, into an array of chars(characters).
            if (!Objects.equals(currentChar, lastChar)) { // If the current Objects don't equal eachother, then the current character is appended to the result array.
                result.append(currentChar); // Add(Append) the current character selected into the result array, if the current requirements/filters are met
            }
            lastChar = currentChar; // 'nuff said, but I will note, this saves time and the possibility of having to plugin pointers.
        }
        return result.toString(); // 'nuff said, but I will note, this is not a System.out.print(), because we're not looking to print out the answer, but instead to turn the result
                                  // from an array into a string, and then return the value under the public String class
    }
}
// Note: 's.toCharArray' can be compaired to Python's 'list(string)' statement

// Your code took 7 milliseconds — faster than 95.29% in Java
